Emergence/Contexts/Presenting Locations/ShowLocationsViewController.swift

```diff
@@ -1,27 +1,70 @@
 import UIKit
 
-class ShowLocationsViewController: UIViewController {
+class ShowLocationsViewController: UIPageViewController, UIPageViewControllerDataSource, UIPageViewControllerDelegate {
     var location: Location!
+    var shows = [Show]()
 
-    override func viewDidLoad() {
-
+    override func awakeFromNib() {
+        delegate = self;
+        dataSource = self;
+        automaticallyAdjustsScrollViewInsets = false;
+    }
 
+    override func viewDidLoad() {
         guard let appVC = self.appViewController else {
             print("you need an app VC")
             return
         }
 
         let network = appVC.context.network
+        let coords = location.coordinates()
+        let showInfo = ArtsyAPI.RunningShowsNearLocation(amount: 2, lat: coords.lat, long: coords.long)
+        network.request(showInfo).mapSuccessfulHTTPToObjectArray(Show).subscribe(next: { decodables in
+            print("decodables: \(decodables)")
+                self.shows = decodables
 
-        let showInfo = ArtsyAPI.UpcomingShowsNearLocation(lat: "", long: "")
-        network.request(showInfo).subscribe(next: { showObject in
-                print(showObject)
+                if self.shows.isEmpty {
+                    // Show a no shows found?
+                } else {
+                    let vcs = [self.viewControllerForIndex(0)!]
+                    self.setViewControllers(vcs, direction: .Forward, animated: false, completion: nil)
+                }
 
-            }, error: { error in
-                print("ERROROR \(error)")
-            }, completed: nil, disposed: nil)
+        }, error: { error in
+            print("ERROROR \(error)")
+        }, completed: nil, disposed: nil)
+    }
+
+    func pageViewController(pageViewController: UIPageViewController, viewControllerBeforeViewController viewController: UIViewController) -> UIViewController? {
+        if shows.count == 1 { return nil }
 
+        let controller = viewController as! ShowViewController
+        let newIndex = (controller.index + 1) % shows.count
+        return viewControllerForIndex(newIndex)
     }
 
+    func pageViewController(pageViewController: UIPageViewController, viewControllerAfterViewController viewController: UIViewController) -> UIViewController? {
+        if shows.count == 1 { return nil }
```

Comment:
this actually made much more sense later with `        guard -1...shows.count ~= index else { return nil }`